R/guess_dates.R

```diff
@@ -0,0 +1,294 @@
+#' Try and guess dates from a characters
```

Comment:
I will dig into this later.

Reply:
This is the primary question that needs to be resolved prior to approving the package for release (given that other critical issues are addressed). We're almost at the finish line 🏁 

What is especially relevant: Are there known cases that this code covers and lubridate/anytime does not? If the code is legacy from Thibault and we can't find the additional coverage, that would be a reason to reduce the complexity here --- and call lubridate/anytime directly where needed. 

Checking this will be incredibly helpful to understand and argue for whatever ends up being included 😊